evalme/image/object_detection.py

```diff
@@ -441,60 +441,72 @@ def distance(self, pred, local_range=1, label_weights=None, per_label=False):
-                    gt_results_rectangle = [item for item in gt_results if item['type'] == 'rectangle']
-                    pred_results_rectangle = [item for item in pred_results if item['type'] == 'rectangle']
-                    score = self._get_max_iou_rectangles(gt_results_rectangle, pred_results_rectangle, threshold)
+                # Tag for region selection [check OCR_geo_tags list]
+                rec_type = None
+                for item in OCREvalItem.OCR_geo_tags:
+                    if item in pred_types and item in gt_types:
+                        rec_type = item
+                # check if both groups have region selection tags
+                if rec_type in pred_types and rec_type in gt_types:
+                    gt_results_rectangle = [item for item in gt_results if item['type'] == rec_type]
+                    pred_results_rectangle = [item for item in pred_results if item['type'] == rec_type]
+                    # get max score for region selection
+                    score = self._get_max_iou_rectangles(gt_results_rectangle, pred_results_rectangle)
                     if score < threshold:
-                        results[id_gt] = 0
+                        if not per_label:
+                            results[id_gt] = 0
                     else:
                         gt_results_labels = [item['value']['labels'] for item in gt_results if item['type'] == 'labels']
                         pred_results_labels = [item['value']['labels'] for item in pred_results if item['type'] == 'labels']
+                        # compare labels
                         if gt_results_labels == pred_results_labels:
-                            gt_results_text = TextAreaEvalItem([item for item in gt_results if item['type'] != 'labels' and item['type'] != 'rectangle'])
-                            pred_results_text = TextAreaEvalItem([item for item in pred_results if item['type'] != 'labels' and item['type'] != 'rectangle'])
+                            gt_results_text = TextAreaEvalItem([item for item in gt_results if item['type'] != 'labels' and item['type'] != rec_type])
+                            pred_results_text = TextAreaEvalItem([item for item in pred_results if item['type'] != 'labels' and item['type'] != rec_type])
+                            # compare text results
                             res = gt_results_text.match(item=pred_results_text, algorithm=algorithm)
                             if per_label:
                                 for item in pred_results_labels:
                                     for subitem in item:
-                                        results[subitem] += res
+                                        results[subitem] += res * score
                                         num_results[subitem] += 1
                             else:
-                                results[id_gt] = res
+                                results[id_gt] = res * score
```

Comment:
when threshold is applied, return only text distance

Reply:
Fixed